src/main/java/uk/gov/pay/webhooks/message/EventMapper.java

```diff
@@ -10,21 +10,30 @@
 import java.util.stream.Collectors;
 
 public class EventMapper {
-    private static final Map<String, EventTypeName> internalToWebhook = Map.of(
-            "PAYMENT_STARTED", EventTypeName.CARD_PAYMENT_STARTED,
-            "AUTHORISATION_SUCCEEDED", EventTypeName.CARD_PAYMENT_SUCCEEDED,
-            "CAPTURE_CONFIRMED", EventTypeName.CARD_PAYMENT_CAPTURED,
-            "REFUND_SUCCEEDED", EventTypeName.CARD_PAYMENT_REFUNDED
+    private static final Map<String, EventTypeName> internalToWebhook = Map.ofEntries(
+            Map.entry("USER_APPROVED_FOR_CAPTURE", EventTypeName.CARD_PAYMENT_SUCCEEDED),
+            Map.entry("SERVICE_APPROVED_FOR_CAPTURE", EventTypeName.CARD_PAYMENT_SUCCEEDED),
+            Map.entry("QUEUED_FOR_CAPTURE", EventTypeName.CARD_PAYMENT_SUCCEEDED),
+            Map.entry("AUTHORISATION_REJECTED", EventTypeName.CARD_PAYMENT_FAILED),
+            Map.entry("AUTHORISATION_CANCELLED", EventTypeName.CARD_PAYMENT_FAILED),
+            Map.entry("GATEWAY_ERROR_DURING_AUTHORISATION", EventTypeName.CARD_PAYMENT_FAILED),
+            Map.entry("GATEWAY_TIMEOUT_DURING_AUTHORISATION", EventTypeName.CARD_PAYMENT_FAILED),
+            Map.entry("UNEXPECTED_GATEWAY_ERROR_DURING_AUTHORISATION", EventTypeName.CARD_PAYMENT_FAILED),
+            Map.entry("CANCELLED_BY_EXTERNAL_SERVICE", EventTypeName.CARD_PAYMENT_FAILED),
+            Map.entry("CANCELLED_BY_USER", EventTypeName.CARD_PAYMENT_FAILED),
+            Map.entry("CANCELLED_BY_EXPIRATION", EventTypeName.CARD_PAYMENT_EXPIRED),
+            Map.entry("CAPTURE_CONFIRMED", EventTypeName.CARD_PAYMENT_CAPTURED),
+            Map.entry("REFUND_SUCCEEDED", EventTypeName.CARD_PAYMENT_REFUNDED)
     );
 
-    private static final Map<EventTypeName, String> webhookToInternal = internalToWebhook
-            .entrySet()
+    private static final Map<EventTypeName, List<String>> webhookToInternal = internalToWebhook
+            .keySet()
             .stream()
-            .collect(Collectors.toMap(Map.Entry::getValue, Map.Entry::getKey));
+            .collect(Collectors.groupingBy(internalToWebhook::get));
 
     private static final List<EventTypeName> childEvents = List.of(EventTypeName.CARD_PAYMENT_REFUNDED);
 
-    public static Optional<String> getInternalEventNameFor(EventTypeName webhookEventTypeName) {
+    public static Optional<List<String>> getInternalEventNameFor(EventTypeName webhookEventTypeName) {
```

Comment:
Two things:

1. Could we return an empty list when there’s nothing rather than an optional?
2. Can this be a set rather than a list?